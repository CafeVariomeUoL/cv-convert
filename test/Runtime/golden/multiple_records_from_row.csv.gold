[
{
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-417",
    "subject_id": "0-0",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-417",
    "subject_id": "0-1",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-417",
    "subject_id": "0-2",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-417",
    "subject_id": "0-3",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-89675",
    "subject_id": "1-0",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-11947",
    "subject_id": "2-0",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-11947",
    "subject_id": "2-1",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-11947",
    "subject_id": "2-2",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-1533312",
    "subject_id": "4-0",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-1533312",
    "subject_id": "4-1",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-228760",
    "subject_id": "5-0",
    "sex": "U"
} , {
    "age": 17,
    "case_type": "suspected",
    "diagnosis": "ORPHA-228760",
    "subject_id": "5-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-417",
    "subject_id": "6-0",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-417",
    "subject_id": "6-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-417",
    "subject_id": "6-2",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-417",
    "subject_id": "6-3",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-417",
    "subject_id": "6-4",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-417",
    "subject_id": "6-5",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-89675",
    "subject_id": "7-0",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-89675",
    "subject_id": "7-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-11947",
    "subject_id": "8-0",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-11947",
    "subject_id": "8-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-11947",
    "subject_id": "8-2",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-0",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-2",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-3",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-4",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-5",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-6",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-48613",
    "subject_id": "9-7",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-1533312",
    "subject_id": "10-0",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-1533312",
    "subject_id": "10-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-1533312",
    "subject_id": "10-2",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-228760",
    "subject_id": "11-0",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "suspected",
    "diagnosis": "ORPHA-228760",
    "subject_id": "11-1",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-228760",
    "subject_id": "11-2",
    "sex": "U"
} , {
    "age": 18,
    "case_type": "confirmed",
    "diagnosis": "ORPHA-228760",
    "subject_id": "11-3",
    "sex": "U"
}
]