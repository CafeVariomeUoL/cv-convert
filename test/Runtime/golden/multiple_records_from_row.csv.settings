{
  "processFunction": "const res = [];\n\nfor(j = 0 ; j < row['Suspected Cases']; j++){\n  res.push({\n    subject_id: `${row.i}-${j}`,\n    sex: \"U\",\n    age: row['Age Group'] === \"<18\" ? 17 : 18,\n    diagnosis: row.Orphacode.replace(\"ORPHA\", \"ORPHA-\"),\n    case_type: \"suspected\"\n  })\n}\n\nfor(j = 0 ; j < row['Confirmed Cases']; j++){\n  res.push({\n    subject_id: `${row.i}-${parseInt(row['Suspected Cases']) + j}`,\n    sex: \"U\",\n    age: row['Age Group'] === \"<18\" ? 17 : 18,\n    diagnosis: row.Orphacode.replace(\"ORPHA\", \"ORPHA-\"),\n    case_type: \"confirmed\"\n  })\n}\n\nreturn res;",
  "libraryFunctions": "Lib.calculate_age = function(dob) { \n    var diff_ms = Date.now() - dob.getTime();\n    var age_dt = new Date(diff_ms); \n  \n    return Math.abs(age_dt.getUTCFullYear() - 1970);\n}\n",
  "tsInterface": "export default interface Row {\n    subject_id:          any;\n    sex:                 \"M\"|\"F\"|\"U\";\n    age:                 number;\n    diagnosis_age?:      number;\n    first_symptoms_age?: number;\n    diagnosis:           string;\n    case_type:           \"suspected\" | \"confirmed\";\n}",
  "jsonSchema": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": {
      "Row": {
        "$ref": "#/definitions/default"
      },
      "default": {
        "title": "default",
        "type": "object",
        "properties": {
          "subject_id": {
            "title": "subject_id"
          },
          "sex": {
            "enum": [
              "F",
              "M",
              "U"
            ],
            "type": "string",
            "title": "sex"
          },
          "age": {
            "type": "number",
            "title": "age"
          },
          "diagnosis_age": {
            "type": "number",
            "title": "diagnosis_age"
          },
          "first_symptoms_age": {
            "type": "number",
            "title": "first_symptoms_age"
          },
          "diagnosis": {
            "type": "string",
            "title": "diagnosis"
          },
          "case_type": {
            "enum": [
              "confirmed",
              "suspected"
            ],
            "type": "string",
            "title": "case_type"
          }
        },
        "additionalProperties": false,
        "required": [
          "age",
          "case_type",
          "diagnosis",
          "sex",
          "subject_id"
        ]
      }
    },
    "$ref": "#/definitions/default"
  }
}