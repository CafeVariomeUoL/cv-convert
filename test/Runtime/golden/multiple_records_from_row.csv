Orphacode,Age Group,Suspected Cases,Confirmed Cases
ORPHA417,<18,1,3
ORPHA89675,<18,0,1
ORPHA11947,<18,3,0
ORPHA48613,<18,0,0
ORPHA1533312,<18,1,1
ORPHA228760,<18,2,0
ORPHA417,>=18,4,2
ORPHA89675,>=18,0,2
ORPHA11947,>=18,2,1
ORPHA48613,>=18,3,5
ORPHA1533312,>=18,2,1
ORPHA228760,>=18,2,2